server/model/telemetry.go

```diff
@@ -0,0 +1,51 @@
+package model
+
+import (
+	"fmt"
+
+	"server/config"
+
+	"github.com/segmentio/analytics-go/v3"
+	"gorm.io/gorm"
+)
+
+// event name - required for Amplitude
+const event string = "Deployment Completed"
+
+// This slice should have the same metrics as /server/model/enums.go DeploymentMetrics
+var metrics = []DeploymentMetric{
+	StartTime,
+	EndTime,
+	ApplicationId,
+	Region,
+	AccessType,
+	DeployStatus,
+	Errors,
+	Retries,
+}
+
+func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
+
+	var propertiesMap = analytics.Properties{}
+	for _, metric := range metrics {
+		propertiesMap[string(metric)] = fmt.Sprintf("%v", Metric(db, metric).MetricValue)
+	}
+	return propertiesMap
+}
+
+func PublishToSegment(db *gorm.DB) {
+
+	client := analytics.New(config.GetEnvironment().SEGMENT_WRITE_KEY)
```

Comment:
This should check for env variables being set. If they aren't, it should do nothing.

Reply:
Added check on env variable